snuba/manual_jobs/__init__.py

```diff
@@ -23,9 +31,15 @@ def __init__(self, job_spec: JobSpec, dry_run: bool) -> None:
         if job_spec.params:
             for k, v in job_spec.params.items():
                 setattr(self, k, v)
+        self._set_status(NOT_STARTED)
 
-    @abstractmethod
     def execute(self) -> None:
+        self._set_status(RUNNING)
+        self._execute()
+        self._set_status(FINISHED)
```

Comment:
I'm thinking that we might want status tracking to live outside `Job`.I think of Job as more of an interface for users to implement, rather than something that contains logic for running jobs itself (I also think things can get confusing quickly when you rely on inheritance in Python). 

What if we had something like a JobRunner static class or module that exposed utilities that would orchestrate the job run, and manage status based on the completely separate Job class? I think that produces a looser coupling.